src/main/java/seedu/address/model/transaction/Date.java

```diff
@@ -0,0 +1,76 @@
+package seedu.address.model.transaction;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Class to store the date of transaction.
+ */
+public class Date {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Date should be in the format DD/MM/YYYY";
+
+    public final String date;
+
+    private final DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+    /**
+     * Constructs a {@code Date}.
+     *
+     * @param dateTime A valid goods name.
+     */
+    public Date(String dateTime) {
+        requireNonNull(dateTime);
+        checkArgument(isValidDate(dateTime), MESSAGE_CONSTRAINTS);
+        this.date = LocalDate.parse(dateTime, dtf).toString();
+    }
+
+    /**
+     * Returns true if a given string is a valid date.
+     */
+    public static boolean isValidDate(String test) {
```

Comment:
Are there no methods in the DateTime java API to do this?

Reply:
agree with the two gentlemen

Reply:
I think just by parsing the date with the given format should already check for the validity of the date! Therefore, the check for valid number of days in the month would not be required.